### ISSUE TITLE
Static analysis result in case of zip(typing.NamedTuple) is different from in case of zip(tuple)

### ISSUE THREAD

**Issue by Kyurenpoto** [microsoft/pylance-release#1658]
## Environment data

Pylance v2021.8.0
-   Language Server version: 2021.8.0 (Pyright 1.1.159)
-   OS and version: Windows 10 (19043.1151)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.5 (pipenv)

## Expected behaviour

No error in following case when use normal tuple:
```python3
x = (["1"], ["2"], [3])
y = list(zip(*x))

for a, b, c in zip(*x):
    print(a, b, c) #1 2 3

```
And Pylance analyzes the type of variables:

![image](https://user-images.githubusercontent.com/18042134/128841230-44841bf0-a02e-4a64-bace-8294d7748e04.png)
![image](https://user-images.githubusercontent.com/18042134/128842811-8c52bcf7-c669-4275-ba96-cdf590eb696f.png)
![image](https://user-images.githubusercontent.com/18042134/128842877-262cd59d-ce73-4d5e-b7f0-6d79452cd6ea.png)
![image](https://user-images.githubusercontent.com/18042134/128842903-2a5161ba-ecc8-4402-8ae7-b897aa55c781.png)

## Actual behaviour

Error occured:
Tuple size mismatch: expected 3 but received 1 Pylance(reportGeneralTypeIssues) [11, 5]

```python3
from typing import NamedTuple

class T(NamedTuple):
    a: list[str]
    b: list[str]
    c: list[int]

x = T(["1"], ["2"], [3])
y = list(zip(*x))

for a, b, c in zip(*x):
    print(a, b, c) #1 2 3

```
And Pylance analyzes the type of variables:

![image](https://user-images.githubusercontent.com/18042134/128843123-51ebff43-cfcc-40f6-9edd-2996091d01db.png)
![image](https://user-images.githubusercontent.com/18042134/128843165-6b7746f9-02ce-4891-8f9a-ee7c80bc9bbf.png)
![image](https://user-images.githubusercontent.com/18042134/128843209-a5f849c4-b417-4bb6-b2f1-3563c19b8979.png)
![image](https://user-images.githubusercontent.com/18042134/128843231-2a2ec5c5-24e5-4d37-ada2-7ad39eec3be8.png)

This is cleary false-positive behavior.

**Comment by erictraut** [comment]
Thanks for reporting the issue. I've added support for unpacking of objects that derive from a known-length tuple. This includes named tuples. The change will be included in the next release of pylance.

**Comment by heejaechang** [comment]
fixed in 2021.8.1 release.
